### ISSUE TITLE
Secrets of type docker-registry are incorrectly generated with 1.9.0 kubectl client

### ISSUE THREAD

**Issue by pierredadt** [kubernetes/kubernetes#61358]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug  
@kubernetes/sig-cli-bugs

**What happened**:

With kubectl client version 1.9.0, in the secrets of the type `docker-registry` generated with `kubectl create secret docker-registry my-secret` the name of the key is `.dockercfg`.
The direct consequence being that the serviceAccounts and other resources using a generated `docker-registry` secret won't be able to pull docker images.

**What you expected to happen**:

In order to match k8s 1.9.0 specifications/documentation, the name of the key in the `docker-registry` secret should be `.dockerconfigjson`

**How to reproduce it (as minimally and precisely as possible)**:

Auto-generate a `docker-registry` type secret with the `kubectl` client in version 1.9.0.
E.g.: 

```
kubectl -n my-namespace create secret docker-registry my-docker-secret \
  --docker-server=my-docker-server-url \
  --docker-username=my-docker-user-id \
  --docker-password=my-docker-user-password \
  --docker-email=my-docker-user-email
```
Edit or get said secret to note that the key is named `.dockercfg` instead of `.dockerconfigjson`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0+coreos.0", GitCommit:"1b69a2a6c01194421b0aa17747a8c1a81738a8dd", GitTreeState:"clean", BuildDate:"2017-12-19T02:52:15Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:

`OpenStack Newton`
- OS (e.g. from /etc/os-release):

`CoreOs`
- Kernel (e.g. `uname -a`):

`Linux node1-public 4.14.19-coreos #1 SMP Wed Feb 14 03:18:05 UTC 2018 x86_64 Intel Core Processor (Broadwell) GenuineIntel GNU/Linux`
- Install tools:

`kubespray [commit fd04c14260605e999a6c1889a1c75239184806d8]`
- Others:

`N/A`

**Comment by liggitt** [comment]
Correct. This was resolved in 1.9.1

/close

**Comment by liggitt** [comment]
See https://github.com/kubernetes/kubernetes/issues/57528
